### PR TITLE
notmuch-mailmover: init at 0.1.0

### PR DESCRIPTION
**Pull request by archer-65**
###### Description of changes

This changes add `notmuch-mailmover`, a tool to move `notmuch` tagged mails into IMAP folders. This package is based on `afew`'s MailMover one, but easier (and more flexible) to configure.
[Homepage](https://github.com/michaeladler/notmuch-mailmover/)
@michaeladler 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by michaeladler** [comment]
Being the author I guess I should maintain this package too if it's accepted :) So feel free to add me to the maintainer list if you like.

**Comment by archer-65** [comment]
> Being the author I guess I should maintain this package too if it's accepted :) So feel free to add me to the maintainer list if you like.

Done! I pinged you appositely :)

**Review by michaeladler — APPROVED** [review]

**Comment by archer-65** [comment]
Should we backport to 22.05?
@michaeladler

**Comment by michaeladler** [comment]
I don't think so. 22.11 is in the currently in the making, so it could end up there if we are lucky or quick enough :)

**Comment by archer-65** [comment]
Any chance to see this package in 22.11?

**Comment by michaeladler** [comment]
Probably best to ping someone with merge permissions on Matrix.

File: pkgs/applications/networking/mailreaders/notmuch/notmuch-mailmover.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (RaitoBezarius) [inline pkgs/applications/networking/mailreaders/notmuch/notmuch-mailmover.nix:1]: This is not correctly formatted, it should be:
```
{ notmuch
, lib
, stdenv
, etc.
}:
```
(please take a look to the other files)

File: pkgs/applications/networking/mailreaders/notmuch/notmuch-mailmover.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "michaeladler";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-b2Q1JcXIp56Niv5kdPgQSM91e8hPPdyhWIG4f7kQn78=";
+  };
+
+  buildInputs = [notmuch];
```
Comment (RaitoBezarius) [inline pkgs/applications/networking/mailreaders/notmuch/notmuch-mailmover.nix:19]: `buildInputs = [ notmuch ];`

File: pkgs/applications/networking/mailreaders/notmuch/notmuch-mailmover.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  cargoSha256 = "sha256-AW0mCdQN3WJhSErJ/MqnNIsRX+C6Pb/zHCQh7v/70MU=";
+
+  meta = with lib; {
+    description = "Application to assign notmuch tagged mails to IMAP folders";
+    homepage = "https://github.com/michaeladler/notmuch-mailmover/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [michaeladler archer-65];
```
Comment (RaitoBezarius) [inline pkgs/applications/networking/mailreaders/notmuch/notmuch-mailmover.nix:27]: `[ michaeladler archer-65 ]`

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Once the minor formatting things are done, I can merge it. :)

**Review by archer-65 — COMMENTED** [review]
Done

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**